jsapp/js/components/library/singleCollectionStore.es6

```diff
@@ -0,0 +1,335 @@
+    actions.library.subscribeToCollection.completed.listen(this.fetchData.bind(this, true));
+    actions.library.unsubscribeFromCollection.completed.listen(this.fetchData.bind(this, true));
+    actions.resources.loadAsset.completed.listen(this.onAssetChanged);
+    actions.resources.updateAsset.completed.listen(this.onAssetChanged);
+    actions.resources.cloneAsset.completed.listen(this.onAssetCreated);
+    actions.resources.createResource.completed.listen(this.onAssetCreated);
+    actions.resources.deleteAsset.completed.listen(this.onDeleteAssetCompleted);
+    // Actions unique to a single collection store (overwriting myLibraryStore)
+    actions.library.searchMyCollectionAssets.started.listen(this.onSearchStarted);
+    actions.library.searchMyCollectionAssets.completed.listen(this.onSearchCompleted);
+    actions.library.searchMyCollectionAssets.failed.listen(this.onSearchFailed);
+    actions.library.searchMyCollectionMetadata.completed.listen(this.onSearchMetadataCompleted);
+
+    // startup store after config is ready
+    actions.permissions.getConfig.completed.listen(this.startupStore);
+  },
+
+  /**
+   * Only makes a call to BE when loaded app on a library route
+   * otherwise wait until route changes to a library (see `onRouteChange`)
+   */
+  startupStore() {
+    if (this.isVirgin && isOnSingleCollectionRoute() && !this.data.isFetchingData) {
+      this.fetchData(true);
+    }
+  },
+
+  setDefaultColumns() {
+    this.data.orderColumnId = this.DEFAULT_ORDER_COLUMN.id;
+    this.data.orderValue = this.DEFAULT_ORDER_COLUMN.defaultValue;
+    this.data.filterColumnId = null;
+    this.data.filterValue = null;
+  },
+
+  // methods for handling search and data fetch
+
+  /**
+   * @return {object} search params shared by all searches
+   */
+  getSearchParams() {
+    const params = {
+      searchPhrase: searchBoxStore.getSearchPhrase(),
+      pageSize: this.PAGE_SIZE,
+      page: this.data.currentPage,
+      uid: getCurrentCollectionUID()
+    };
+
+    if (this.data.filterColumnId !== null) {
+      const filterColumn = ASSETS_TABLE_COLUMNS.get(this.data.filterColumnId);
+      params.filterProperty = filterColumn.filterBy;
+      params.filterValue = this.data.filterValue;
+    }
+
+    return params;
+  },
+
+  fetchMetadata() {
+    actions.library.searchMyCollectionMetadata(this.getSearchParams());
+  },
+
+  /**
+   * @param {boolean} needsMetadata
+   */
+  fetchData(needsMetadata = false) {
+    actions.library.searchMyCollectionMetadata(this.getSearchParams());
+    if (this.abortFetchData) {
+      this.abortFetchData();
+    }
+
+    const params = this.getSearchParams();
+    // Surrounds `filterValue` with double quotes to avoid filters that have
+    // spaces which would split the query in two, thus breaking the filter
+    if (params.filterProperty !== undefined) {
+      params.filterValue = JSON.stringify(params.filterValue); // Adds quotes
+    }
```

Comment:
Please add the same double quotes improvement to `publicCollectionsStore.es6` and `myLibraryStore.es6`